docs/QATv2.md

```diff
@@ -0,0 +1,60 @@
+
+
+## Scale Adjustments
+
+To implement the threshold-based quantization, the scales of the layers are adjusted. The scales are adjusted based on the type of operation that is performed on the layers. The scale adjustments are made for residual additions, concatenations, and layer sharing. In the f
+
+Figure 1. shows the scale adjustments for residual additions. In the figure, Layer1 and Layer2 are layers that are added together. The scale of the residual addition is selected as the scale of the layers that are connected to the residual addition.
+
+<img src="QATv2-Adds.png" style="zoom: 50%;" />
+
+Figure 1. Scale Adjustments for Residual Additions
+
+Figure 2. shows the scale adjustments for concatenations. In the figure, Layer1 and Layer2 are layers that are concatenated. The maximum scale of the layers is selected as the scale for the concatenated layer.
+
+<img src="QATv2-Concats.png" style="zoom: 50%;" />
+
+Figure 2. Scale Adjustments for Concatenations
+
+Figure 3. shows the scale adjustments for layer sharing. In the figure, Layer1, Layer2 and Layer3 are layers that share weights. The maximum scale of the layers is selected as the scale for the shared layer.
+
+<img src="QATv2-Layer Sharing.png" style="zoom: 50%;" />
+
+Figure 3. Scale Adjustments for Layer Sharing
+
+Figure 4. provides a simplified diagram showing how the scaling-down and scale carry-over operations are implemented. In the diagram, Layer1 and Layer2 represent linear layers with weights w1 and w2, and biases b1 and b2. S1 and S2 represent the activation scales, which are calculated as previously described. As shown, the output of Layer1 is scaled down using the S1 threshold, and the scale carry-over operation is achieved by adjusting
+Layer2’s scale and dividing its biases accordingly.
+
+<img src="QATv2-Apply Scales.png" style="zoom: 50%;" />
+
+Figure 4. Scaling-down and Scale Carry Over Diagram
+
+## Weights Quantization
+
+After determining the activation thresholds and scales, the next step is to quantize the weights. The weights are quantized using the QAT framework, which is based on the method proposed by Jacob et al. [2]. While training the model, weights and biasses are fake quantized to 8-bit integers. The fake quantization is done by quantizing the weights and biases to 8-bit integers and then dequantizing them back to floating-point numbers.
```

Comment:
weights are not necessarily quantized to 8-bits.

Reply:
Resolved in the latest commit.